### PR TITLE
[Docs] Translate the logging

### PR DESCRIPTION
**Pull request by HAOCHENYE**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/advanced_tutorials/logging.md [hunk 1]
```diff
@@ -1,3 +1,295 @@
...
+)
+runner.train()
+```
+
+```
+08/21 02:58:41 - mmengine - INFO - Epoch(train) [1][10/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0019  data_time: 0.0004  loss1: 0.8381  loss2: 0.9007  loss: 1.7388
+08/21 02:58:41 - mmengine - INFO - Epoch(train) [1][20/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0029  data_time: 0.0010  loss1: 0.1978  loss2: 0.4312  loss: 0.6290
+```
+
+LogProcessor will output the log as the following statistic method:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/logging.md:57]:
```suggestion
LogProcessor will output the log as the following format:
```

File: docs/en/advanced_tutorials/logging.md [hunk 2]
```diff
@@ -1,3 +1,295 @@
...
+Output the value of all custom logsthe at last iteration by default.
+```
+
+```{warnning}
+log_processor output the epoch based log by default(`by_epoch=True`). To get a expected log matched with the `train_cfg`, we should set the same value for `by_epoch` in `train_cfg` and `log_processor`.
+```
+
+Based on the rules above, the sample code will statistic the average value of the `loss1` and `loss2` every 10 iterations.
+
+If we want to statistic the global average value of `loss1`, we can set `custom_cfg` like this:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/logging.md:83]:
```suggestion
If we want to count the global average value of `loss1`, we can set `custom_cfg` like this:
```

File: docs/en/advanced_tutorials/logging.md [hunk 3]
```diff
@@ -1,3 +1,295 @@
...
+)
+runner.train()
+```
+
+```
+08/21 02:58:49 - mmengine - INFO - Epoch(train) [1][10/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0026  data_time: 0.0007  loss1: 0.7381  loss2: 0.8446  loss: 1.5827
+08/21 02:58:49 - mmengine - INFO - Epoch(train) [1][20/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0030  data_time: 0.0012  loss1: 0.4521  loss2: 0.3939  loss: 0.5600
+```
+
+`data_src` means the original loss name, `method_name` means the statistic method, `window_size` means the window size of the statistic method. Since we want to statistic the global average value of `loss1`, we set `window_size` to `global`.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/logging.md:106]:
```suggestion
`data_src` means the original loss name, `method_name` means the statistic method, `window_size` means the window size of the statistic method. Since we want to count the global average value of `loss1`, we set `window_size` to `global`.
```

File: docs/en/advanced_tutorials/logging.md [hunk 4]
```diff
@@ -1,3 +1,295 @@
...
+
+Currently, MMEngine support the following statistic methods:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/logging.md:108]:
```suggestion
Currently, MMEngine supports the following statistic methods:
```

File: docs/en/advanced_tutorials/logging.md (hunk 3 above)
Comment (C1rN09) [reply docs/en/advanced_tutorials/logging.md:106]: Seems more below

File: docs/en/advanced_tutorials/logging.md [hunk 5]
```diff
@@ -1,3 +1,295 @@
...
+```
+08/21 18:16:22 - mmengine - DEBUG - Get class `LocalVisBackend` from "vis_backend" registry in "mmengine"
+08/21 18:16:22 - mmengine - DEBUG - An `LocalVisBackend` instance is built from registry, its implementation can be found in mmengine.visualization.vis_backend
+08/21 18:16:22 - mmengine - DEBUG - Get class `RuntimeInfoHook` from "hook" registry in "mmengine"
+08/21 18:16:22 - mmengine - DEBUG - An `RuntimeInfoHook` instance is built from registry, its implementation can be found in mmengine.hooks.runtime_info_hook
+08/21 18:16:22 - mmengine - DEBUG - Get class `IterTimerHook` from "hook" registry in "mmengine"
+...
+```
+
+Besides, logs of different ranks will be save in `debug` mode. It you training your model with the shared storage. The heriarchy of the log is as follows:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/logging.md:256]:
```suggestion
Besides, logs of different ranks will be saved in `debug` mode if you are training your model with the shared storage. The heriarchy of the log is as follows:
```

File: docs/en/advanced_tutorials/logging.md [hunk 6]
```diff
@@ -1,3 +1,295 @@
...
+├── tmp_rank5.log
+├── tmp_rank6.log
+└── tmp_rank7.log
+...
+└── tmp_rank63.log
+```
+
+The log of Multi-machine with independent storage:
+
+多机多卡，独立存储的情况：
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/logging.md:274]: Miss translation

File: docs/en/advanced_tutorials/logging.md (hunk 2 above)
Comment (HAOCHENYE) [reply docs/en/advanced_tutorials/logging.md:83]: d

**Review by C1rN09 — APPROVED** [review]

File: docs/en/advanced_tutorials/logging.md [hunk 7]
```diff
@@ -1,3 +1,293 @@
 # Logging
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/logging.html).
+[Runner](../tutorials/runner.md) will produce a lot of logs during the running process, such as loss, iteration time, learning rate, etc. MMEngine implements a flexible logging system that allows us to choose different types of log statistical method when configuring the runner. It could help us set/get the recorded log at any location in the code.
+
+## Flexible Logging System
+
+Logging systom is configured by passing a [LogProcessor](mmengine.logging.LogProcessor) to the runner. If no log processor is passed, the runner will use the default log processor, which is equivalent to:
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:7]:
```suggestion
Logging system is configured by passing a [LogProcessor](mmengine.logging.LogProcessor) to the runner. If no log processor is passed, the runner will use the default log processor, which is equivalent to:
```

File: docs/en/advanced_tutorials/logging.md [hunk 8]
```diff
@@ -1,3 +1,293 @@
...
+
+```python
+log_processor = dict(window_size=10, by_epoch=True, custom_cfg=None, num_digits=4)
+```
+
+```python
+log_processor = dict(window_size=10, by_epoch=True, custom_cfg=None, num_digits=4)
+```
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:15]:
```suggestion

```python
log_processor = dict(window_size=10, by_epoch=True, custom_cfg=None, num_digits=4)
```
```

File: docs/en/advanced_tutorials/logging.md [hunk 9]
```diff
@@ -1,3 +1,293 @@
...
+08/21 02:58:41 - mmengine - INFO - Epoch(train) [1][10/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0019  data_time: 0.0004  loss1: 0.8381  loss2: 0.9007  loss: 1.7388
+08/21 02:58:41 - mmengine - INFO - Epoch(train) [1][20/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0029  data_time: 0.0010  loss1: 0.1978  loss2: 0.4312  loss: 0.6290
+```
+
+LogProcessor will output the log as the following format:
+
+- The prefix of the log:
+  - epoch mode(`by_epoch=True`): `Epoch(train) [{current_epoch}/{current_iteration}]/{dataloader_length}`
+  - iteration mode(`by_epoch=False`): `Iter(train) [{current_iteration}/{max_iteration}]`)
+- Learning rate(`lr`): The learing rate of the last iteration.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:62]:
```suggestion
- Learning rate (`lr`): The learning rate of the last iteration.
```

File: docs/en/advanced_tutorials/logging.md [hunk 10]
```diff
@@ -1,3 +1,293 @@
...
+)
+runner.train()
+```
+
+```
+08/21 02:58:41 - mmengine - INFO - Epoch(train) [1][10/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0019  data_time: 0.0004  loss1: 0.8381  loss2: 0.9007  loss: 1.7388
+08/21 02:58:41 - mmengine - INFO - Epoch(train) [1][20/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0029  data_time: 0.0010  loss1: 0.1978  loss2: 0.4312  loss: 0.6290
+```
+
+LogProcessor will output the log as the following format:
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:57]:
```suggestion
LogProcessor will output the log in the following format:
```

File: docs/en/advanced_tutorials/logging.md [hunk 11]
```diff
@@ -1,3 +1,293 @@
 # Logging
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/logging.html).
+[Runner](../tutorials/runner.md) will produce a lot of logs during the running process, such as loss, iteration time, learning rate, etc. MMEngine implements a flexible logging system that allows us to choose different types of log statistical method when configuring the runner. It could help us set/get the recorded log at any location in the code.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:3]:
```suggestion
[Runner](../tutorials/runner.md) will produce a lot of logs during the running process, such as loss, iteration time, learning rate, etc. MMEngine implements a flexible logging system that allows us to choose different types of log statistical methods when configuring the runner. It could help us set/get the recorded log at any location in the code.
```

File: docs/en/advanced_tutorials/logging.md [hunk 12]
```diff
@@ -1,3 +1,293 @@
...
+```{note}
+`window_size=10` by default.
+
+The significant digits(`num_digits`) of the log is 4 by default.
+
+Output the value of all custom logsthe at last iteration by default.
+```
+
+```{warnning}
+log_processor output the epoch based log by default(`by_epoch=True`). To get a expected log matched with the `train_cfg`, we should set the same value for `by_epoch` in `train_cfg` and `log_processor`.
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:78]:
```suggestion
log_processor outputs the epoch based log by default(`by_epoch=True`). To get an expected log matched with the `train_cfg`, we should set the same value for `by_epoch` in `train_cfg` and `log_processor`.
```

File: docs/en/advanced_tutorials/logging.md [hunk 13]
```diff
@@ -1,3 +1,293 @@
...
+```
+
+```
+08/21 03:17:26 - mmengine - INFO - Epoch(train) [1][10/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0021  data_time: 0.0006  loss1: 1.8495  loss2: 1.3427  loss: 3.1922  loss1_local_max: 2.8872  loss1_global_max: 2.8872
+08/21 03:17:26 - mmengine - INFO - Epoch(train) [1][20/25]  lr: 1.0000e-02  eta: 0:00:00  time: 0.0024  data_time: 0.0010  loss1: 0.5464  loss2: 0.7251  loss: 1.2715  loss1_local_max: 2.8872  loss1_global_max: 2.8872
+```
+
+More examples can be found in [log_processor](mmengine.logging.LogProcessor).
+
+## Custom log
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:177]:
```suggestion
## Customize log
```

File: docs/en/advanced_tutorials/logging.md [hunk 14]
```diff
@@ -1,3 +1,293 @@
...
+
+The logging system could not only log the `loss`, `lr`, .etc, but can also collect and output the the custom log. For example, if we want to statistic the intermediate `loss`:
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:179]:
```suggestion
The logging system could not only log the `loss`, `lr`, .etc but also collect and output the custom log. For example, if we want to statistic the intermediate `loss`:
```

File: docs/en/advanced_tutorials/logging.md [hunk 15]
```diff
@@ -1,3 +1,293 @@
...
+```
+08/21 18:16:22 - mmengine - DEBUG - Get class `LocalVisBackend` from "vis_backend" registry in "mmengine"
+08/21 18:16:22 - mmengine - DEBUG - An `LocalVisBackend` instance is built from registry, its implementation can be found in mmengine.visualization.vis_backend
+08/21 18:16:22 - mmengine - DEBUG - Get class `RuntimeInfoHook` from "hook" registry in "mmengine"
+08/21 18:16:22 - mmengine - DEBUG - An `RuntimeInfoHook` instance is built from registry, its implementation can be found in mmengine.hooks.runtime_info_hook
+08/21 18:16:22 - mmengine - DEBUG - Get class `IterTimerHook` from "hook" registry in "mmengine"
+...
+```
+
+Besides, logs of different ranks will be saved in `debug` mode if you are training your model with the shared storage. The heriarchy of the log is as follows:
```
Comment (zhouzaida) [inline docs/en/advanced_tutorials/logging.md:256]:
```suggestion
Besides, logs of different ranks will be saved in `debug` mode if you are training your model with the shared storage. The hierarchy of the log is as follows:
```

**Review by zhouzaida — APPROVED** [review]
